### PR TITLE
Update sample copy

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

The way gem2s params is calculated is changed in API PR 188. This PR updates the mechanism accordingly.

#### Link to staging deployment URL 

N.A.

#### Links to any Pull Requests related to this

https://github.com/biomage-ltd/api/pull/188

#### Anything else the reviewers should know about the changes here

To test : 
- Get an experiment from production that you want to copy. Note the experiment id.
- Setup a staging environment
- Run `biomage experiment copy -e {experiment_id}`
- Once the copying is done, open your staging environment. Find the copied project
- Clicking "Launch Analysis" should not re-run GEM2S.

Note : 
Currently there is a 5GB limit of the file that we can copy. This due to the limitation of the `copy_object`  method that we use (https://docs.aws.amazon.com/cli/latest/reference/s3api/copy-object.html). Amazon recommends the use of multipart uploads to copy larger files : https://docs.aws.amazon.com/AmazonS3/latest/userguide/CopyingObjectsMPUapi.html. This is not implemented in this ticket.

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] New commands have been added to Makefile/test
- [ ] Unit tests written
- [X] Run make test
- [X] Test any modified command

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: biomage/experiment/utils.py [hunk 1]
```diff
@@ -158,30 +158,43 @@ def create_gem2s_hash(experiment, project, samples):
     if experiment['meta']['M']['organism'].get('S'):
         organism = experiment['meta']['M']['organism']['S']
 
-    sample_ids = [sample_id for sample_id in samples['M']]
+    # Filter 'ids' key which is present in older samples object
+    unsorted_sample_ids = [
+        sample_id
+        for sample_id
+        in samples['M']
+        if sample_id != 'ids'
```
Comment (cosa65) [inline biomage/experiment/utils.py:166]: The line splitting here is kind of weird and hard to read, could you leave it a little more like it was originally?

File: biomage/utils/data.py [hunk 2]
```diff
@@ -83,12 +83,6 @@ def copy_s3_files(sandbox_id, prefix, source_bucket, target_bucket):
         experiment_id = obj["Key"].split("/")[0]
         target_key = obj["Key"].replace(experiment_id, f"{sandbox_id}-{experiment_id}")
 
-        # biomage-originals- uses projectId/sampleId/file schema,
-        # so sampleId has to be prefixed too
-        if 'biomage-originals-' in target_bucket:
-            sample_id = obj["Key"].split("/")[1]
-            target_key = target_key.replace(sample_id, f"{sandbox_id}-{sample_id}")
-
```
Comment (cosa65) [inline biomage/utils/data.py:91]: Why was this deleted?

File: biomage/experiment/utils.py (hunk 1 above)
Comment (aerlaut) [reply biomage/experiment/utils.py:166]: The linter complains because the line is too long. Does this change make it make easier to read?

File: biomage/utils/data.py (hunk 2 above)
Comment (aerlaut) [reply biomage/utils/data.py:91]: The deleted line of code prefixes the sample ids in the S3 sample file path (schema : projectUuid/sampleUuid/filename) when copying files. However, when copying the experiment data, we do not prefix sample ids. We do not prefix sampleIds because we also copy the user's processed Seurat object, which contains unprefixed sample object. If we prefix sample-ids, we will have problem accessing the data in the Seurat object.

**Review by cosa65 — APPROVED** [review]
